ui/ducks/send/send.js

```diff
@@ -949,28 +1036,37 @@ const slice = createSlice({
      * @returns {void}
      */
     updateAsset: (state, action) => {
-      const { asset, initialAssetSet } = action.payload;
+      const { asset, initialAssetSet, isReceived } = action.payload;
       const draftTransaction =
         state.draftTransactions[state.currentTransactionUUID];
 
-      draftTransaction.asset.type = asset.type;
-      draftTransaction.asset.balance = asset.balance;
-      draftTransaction.asset.error = asset.error;
+      const targetAsset =
+        draftTransaction[isReceived ? 'receiveAsset' : 'sendAsset'];
+
+      targetAsset.type = asset.type;
+      targetAsset.balance = asset.balance;
+      targetAsset.error = asset.error;
 
       if (
-        draftTransaction.asset.type === AssetType.token ||
-        draftTransaction.asset.type === AssetType.NFT
+        targetAsset.type === AssetType.token ||
+        targetAsset.type === AssetType.NFT
       ) {
-        draftTransaction.asset.details = asset.details;
+        targetAsset.details = asset.details;
       } else {
         // clear the details object when sending native currency
-        draftTransaction.asset.details = null;
+        targetAsset.details = null;
         if (draftTransaction.recipient.error === CONTRACT_ADDRESS_ERROR) {
           // Errors related to sending tokens to their own contract address
           // are no longer valid when sending native currency.
           draftTransaction.recipient.error = null;
         }
       }
+
+      // if the send asset is set, match the receive asset
+      if (!isReceived) {
+        draftTransaction.receiveAsset = targetAsset;
+      }
+
```

Comment:
Should we preserve receiveAsset if it's not the first time a sendAsset is being selected?

```suggestion
      // if the send asset is set, match the receive asset
      if (!isReceived && initialAssetSet) {
        draftTransaction.receiveAsset = targetAsset;
      }

```

Reply:
The naming is a bit misleading. It seems like this is only called with `initialAssetSet` when returning from the confirmation screen (i.e., EDIT stage)